src/NativeShell/src/AppPreferences.java

```diff
@@ -28,4 +28,24 @@ public void setIgnoreBatteryOptimizations(boolean value) {
         editor.putBoolean(context.getString(R.string.pref_ignore_battery_optimizations), value);
         editor.apply();
     }
+
+    public Boolean getDownloadMethodDialogShown() {
+        return sharedPreferences.getBoolean(context.getString(R.string.pref_download_method_dialog_shown), false);
```

Comment:
You should not translate the keys in SharedPreferences. When a user changes the language all settings will be gone because the key changed.

Reply:
I forgot to add the strings themselves, but these would of course be untranslatable when added to the values. This is a pattern I picked up a while back because it helps if you need to reference the keys in more than one location so you have a source of truth for the keys. I think it has value.

Reply:
That makes sense, I assumed they would be translatable. My bad